qiskit_experiments/data_processing/data_processor.py

```diff
@@ -0,0 +1,88 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Class that ties together actions on the data."""
+
+from typing import Any, Dict, List, Union
+
+from qiskit_experiments.data_processing.base import DataAction
+from qiskit_experiments.data_processing.exceptions import DataProcessorError
+
+
+class DataProcessor:
+    """
+    Defines the actions done on the measured data to bring it in a form usable
+    by the calibration analysis classes.
+    """
+
+    def __init__(self):
```

Comment:
This should take a list of node as input argument and call append in the constructor for validation. This is convenient to prepare preset processors, i.e.

```python
class Level1Processor(DataProcessor):
   def __init__():
      chain = [Kernel(), Discriminator(), Population()]
      super().__init__(*chain)
```

Reply:
Good idea. See here: 93cdfae